### PR TITLE
[SPARK-42301][SQL] Assign name to _LEGACY_ERROR_TEMP_1129

### PR DESCRIPTION
**Pull request by itholic**
### What changes were proposed in this pull request?

This PR proposes to assign name to _LEGACY_ERROR_TEMP_1129, "UNABLE_TO_INFER_SCHEMA".

### Why are the changes needed?

We should assign proper name to _LEGACY_ERROR_TEMP_*

### Does this PR introduce _any_ user-facing change?

No

### How was this patch tested?

`./build/sbt "sql/testOnly org.apache.spark.sql.SQLQueryTestSuite*"`

### CODE REVIEW

**Comment by itholic** [comment]
cc @srielau @MaxGekk @cloud-fan Please review this when you find some time

File: core/src/main/resources/error/error-classes.json [hunk 1]
```diff
@@ -1328,6 +1328,12 @@
       "Unable to convert SQL type <toType> to Protobuf type <protobufType>."
     ]
   },
+  "UNABLE_TO_INFER_SCHEMA" : {
+    "message" : [
+      "Unable to infer schema for <format>. It must be specified manually."
+    ],
+    "sqlState" : "46110"
```
Comment (srielau) [inline core/src/main/resources/error/error-classes.json:1335]:
```suggestion
    "sqlState" : "42KD9"
```

**Review by srielau — APPROVED** [review]
Assigned SQLSTATE

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Updated, thanks!

Btw, seems like we don't have "42KD9" in https://github.com/apache/spark/tree/master/core/src/main/resources/error.

Do we want to update the list ?

**Comment by itholic** [comment]
Test fails since we don't have 42KD9 in SQLSTATE lists.

Please refer to https://github.com/apache/spark/pull/39886.

File: core/src/main/resources/error/error-classes.json (hunk 1 above)
Comment (srielau) [reply core/src/main/resources/error/error-classes.json]: Yes
https://docs.google.com/spreadsheets/d/1hrQBSuHooiozUNAQTHiYq3WidS1uliHpl9cYfWpig1c/edit?usp=sharing

We should sync once in a while

Comment (itholic) [reply core/src/main/resources/error/error-classes.json]: Got it. Let me followup the list periodically.
Thanks! :-)

**Comment by itholic** [comment]
Let me rebase to master once https://github.com/apache/spark/pull/39886 get merged.

**Review by MaxGekk — APPROVED** [review]
Waiting for CI.

**Comment by MaxGekk** [comment]
+1, LGTM. Merging to master/3.4.
Thank you, @itholic and @srielau for review.

**Pull request closed without merging**